hazelcast-jet-core/src/test/java/com/hazelcast/jet/pipeline/BatchStageTest.java

```diff
@@ -870,4 +876,74 @@ public void customTransform_keyed() {
         // multiple processors will observe the same keys and the counts won't match.
         assertEquals("0\n1", streamToString(sinkStreamOf(Integer.class), Object::toString));
     }
+
+    @Test
+    public void batchAddTimestampPreserveOrderTest() {
+        // Given
+        List<Integer> items = IntStream.range(0, 10).boxed().collect(toList());
+        BatchSource<Integer> source = TestSources.items(items);
+
+        // When
+        p.drawFrom(source)
+         .addTimestamps(o -> 0L, 0)
+         .drainTo(assertOrdered(items));
+
+        // Then
+        execute();
+    }
+
+    @Test
+    public void batchAddTimestampMatchingUpstreamParallelism_whenUpstreamHavePreferredLocalParallelism() {
```

Comment:
`addTimestamps_whenUpstreamHavePreferredLocalParallelism_thenMatchUpstream` or similar for other tests

Reply:
`addTimestamps_when_upstreamHasPreferredLocalParallelism_then_lpMatchUpstream`